src/renderer/entities/wallet/lib/permission-utils.ts

```diff
@@ -0,0 +1,143 @@
+  isRejectMultisigTxAvailable,
+  isCreateAnyProxyAvailable,
+  isCreateNonAnyProxyAvailable,
+  isRemoveProxyAvailable,
+};
+
+function isTransferAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  if (walletUtils.isWatchOnly(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isProxied(wallet)) {
+    const account = accounts[0] as ProxiedAccount;
+
+    return walletUtils.isAnyProxyType(account);
+  }
+
+  return true;
+}
+
+function isReceiveAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  return !walletUtils.isWatchOnly(wallet);
+}
+
+function isStakingAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  if (walletUtils.isWatchOnly(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isProxied(wallet)) {
+    const account = accounts[0] as ProxiedAccount;
+
+    return (
+      walletUtils.isAnyProxyType(account) ||
+      walletUtils.isStakingProxyType(account) ||
+      walletUtils.isNonTransferProxyType(account)
+    );
+  }
+
+  return true;
+}
+function isCreateMultisigTxAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  if (walletUtils.isWatchOnly(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isMultisig(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isProxied(wallet)) {
+    const account = accounts[0] as ProxiedAccount;
+
+    return walletUtils.isAnyProxyType(account) || walletUtils.isNonTransferProxyType(account);
+  }
+
+  return true;
+}
+function isApproveMultisigTxAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  if (walletUtils.isWatchOnly(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isMultisig(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isProxied(wallet)) {
+    const account = accounts[0] as ProxiedAccount;
+
+    return walletUtils.isAnyProxyType(account) || walletUtils.isNonTransferProxyType(account);
+  }
+
+  return true;
+}
+function isRejectMultisigTxAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  if (walletUtils.isWatchOnly(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isMultisig(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isProxied(wallet)) {
+    const account = accounts[0] as ProxiedAccount;
+
+    return walletUtils.isAnyProxyType(account) || walletUtils.isNonTransferProxyType(account);
+  }
+
+  return true;
+}
+
+function isCreateAnyProxyAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  if (walletUtils.isWatchOnly(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isProxied(wallet)) {
+    const account = accounts[0] as ProxiedAccount;
+
+    return walletUtils.isAnyProxyType(account);
+  }
+
+  return true;
+}
+
+function isCreateNonAnyProxyAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  if (walletUtils.isWatchOnly(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isProxied(wallet)) {
+    const account = accounts[0] as ProxiedAccount;
+
+    return walletUtils.isAnyProxyType(account) || walletUtils.isNonTransferProxyType(account);
```

Comment:
`proxyType` belongs to account, so it's correct to call `accountUtils. isAnyProxyType(...)`